src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.ViewSpanChangedEventSource.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Tagging
+{
+    internal partial class TaggerEventSources
+    {
+        private class ViewSpanChangedEventSource : ITaggerEventSource
+        {
+            private readonly ForegroundThreadAffinitizedObject foregroundObject = new ForegroundThreadAffinitizedObject();
+
+            private readonly ITextView textView;
+            private readonly TaggerDelay textChangeDelay;
+            private readonly TaggerDelay scrollChangeDelay;
+
+            private Span? span;
+            private ITextSnapshot viewTextSnapshot;
+            private ITextSnapshot viewVisualSnapshot;
+
+            public event EventHandler<TaggerEventArgs> Changed;
+            public event EventHandler UIUpdatesPaused { add { } remove { } }
+            public event EventHandler UIUpdatesResumed { add { } remove { } }
+
+            public ViewSpanChangedEventSource(ITextView textView, TaggerDelay textChangeDelay, TaggerDelay scrollChangeDelay)
+            {
+                Debug.Assert(textView != null);
+                this.textView = textView;
+                this.textChangeDelay = textChangeDelay;
+                this.scrollChangeDelay = scrollChangeDelay;
+            }
+
+            public void Connect()
+            {
+                foregroundObject.AssertIsForeground();
+                textView.LayoutChanged += OnLayoutChanged;
+            }
+
+            public void Disconnect()
+            {
+                foregroundObject.AssertIsForeground();
+                textView.LayoutChanged -= OnLayoutChanged;
```

Comment:
I debugged through and teh order of events is:

textView closes.  taggers get released.  last ref count on the tagger goes away.  tagger disconnects from event sources.  we disconnect from layout changed.

So i don't see much value in preemptively doing that since it's going to happen anyways.